cli/pcluster/config/pcluster_config.py

```diff
@@ -439,28 +440,32 @@ def get_compute_availability_zone(self):
 
     def __load_json_config(self, cfn_params):
         """Retrieve Json configuration params from the S3 bucket linked from the cfn params."""
-        s3_bucket_name = get_cfn_param(cfn_params, "ResourcesS3Bucket")
-        if s3_bucket_name and s3_bucket_name != "NONE":
-            tmp_s3_config_path = os.path.join(mkdtemp(), "s3_config.json")
-            try:
-                s3 = boto3.client("s3")
-                s3.download_file(s3_bucket_name, "configs/cluster-config.json", tmp_s3_config_path)
-
-                with open(tmp_s3_config_path) as s3_config_file:
-                    return json.load(s3_config_file, object_pairs_hook=OrderedDict)
-            except ClientError as e:
-                self.error(
-                    "Unable to retrieve configuration from bucket '{0}'.\n{1}".format(
-                        s3_bucket_name, e.response.get("Error").get("Message")
+        json_config = None
+        scheduler = get_cfn_param(cfn_params, "Scheduler")
+        if is_hit_enabled_cluster(scheduler):
+            s3_bucket_name = get_cfn_param(cfn_params, "ResourcesS3Bucket")
+            if s3_bucket_name and s3_bucket_name != "NONE":
```

Comment:
don't we need to also fail with error if s3_bucket_name is not found or NONE?

Reply:
I don't think so. This would be the case of "sge" or "torque" schedulers, for which we do not create the resources bucket. Am I missing something?

Reply:
As per our conversation, updated code to use read_remote_file and added unit tests